addons/crm/models/res_partner.py

```diff
@@ -38,9 +38,27 @@ def default_get(self, fields):
 
     @api.multi
     def _compute_opportunity_count(self):
+        result_list = self.env['crm.lead'].read_group(
+            [
+                ('type', '=', 'opportunity'),
+                '|',
+                ('partner_id.commercial_partner_id', 'in', self.ids),
```

Comment:
`child_of` is needed, as you can have more than 2 levels.

Reply:
Solved, cross-posting answer from https://github.com/odoo/odoo/pull/71950#issuecomment-858553512